### PR TITLE
Docs: Distinguish examples in rules under Stylistic Issues part 6

### PR DESCRIPTION
**Pull request by scriptdaemon**
Found some time to get a quick batch in.

`object-curly-newline` has the option to have separate configurations for object literals and destructuring assignments. My mind is drawing a blank, did we have any specific way to handle options for this kind of case?

@pedrottimark Mind giving this a look?

Slowly and surely... :smiley:

### CODE REVIEW

File: docs/rules/no-trailing-spaces.md [hunk 1]
```diff
@@ -1,12 +1,14 @@
...
 
 (fixable) The `--fix` option on the [command line](../user-guide/command-line-interface#fix) automatically fixes problems reported by this rule.
 
-Sometimes in the course of editing files, you can end up with extra whitespace at the end of lines. These whitespace differences can be picked up by source control systems and flagged as diffs, causing frustration for developers. While this extra whitespace causes no functional issues, many code conventions require that trailing spaces be removed before checkin.
+Sometimes in the course of editing files, you can end up with extra whitespace at the end of lines. These whitespace differences can be picked up by source control systems and flagged as diffs, causing frustration for developers. While this extra whitespace causes no functional issues, many code conventions require that trailing spaces be removed before check-in.
 
 ## Rule Details
 
-The following patterns are considered problems:
+This rule disallow trailing whitespace at the end of lines.
```
Comment (pedrottimark) [inline docs/rules/no-trailing-spaces.md]: s/disallow/disallows/

Comment (pedrottimark) [inline docs/rules/no-trailing-spaces.md]: What do you think about deleting the comment in incorrect code below and adding it here?

This rule disallows trailing whitespace (spaces, tabs, and other Unicode whitespace characters) at the end of lines.

File: docs/rules/no-trailing-spaces.md [hunk 2]
```diff
@@ -17,7 +19,7 @@ var foo = 0;//•••••
 var baz = 5;//••
```
Comment (pedrottimark) [inline docs/rules/no-trailing-spaces.md]: Suggest copy `//••••` from correct code for the option below to a line between the two variable declarations in incorrect code here.

File: docs/rules/no-underscore-dangle.md [hunk 3]
```diff
@@ -53,6 +35,17 @@ obj.__proto__ = {};
 var file = __filename;
 ```
 
+## Options
+
+This rule has an object option:
+
+* `"allow"` allows specified names to be used with a dangling underscore
```
Comment (pedrottimark) [inline docs/rules/no-underscore-dangle.md]: allows specified identifiers to have dangling underscores

File: docs/rules/no-unneeded-ternary.md [hunk 4]
```diff
@@ -47,11 +43,7 @@ var a = x === 2 ? true : false;
...
 ```
 
-The following pattern is considered a warning when `defaultAssignment` is `false`:
-
-```js
-var a = x ? x : 1;
-```
+Examples of **correct** code for this rule:
 
 The following patterns are not considered problems:
```
Comment (pedrottimark) [inline docs/rules/no-unneeded-ternary.md]: delete this paragraph :)

File: docs/rules/no-unneeded-ternary.md [hunk 5]
```diff
@@ -65,11 +57,24 @@ var a = x !== false;
...
+## Options
+
+This rule has an object option:
+
+* `"defaultAssignment": true` (default) allows the conditional expression as a default assignment pattern
+* `"defaultAssignment": false` disallows the conditional expression as a default assignment pattern
+
+### defaultAssignment
+
+Examples of **incorrect** code for this rule with the `{ "defaultAssignment": false }` option:
```
Comment (pedrottimark) [inline docs/rules/no-unneeded-ternary.md]: Examples of additional **incorrect** code …

File: docs/rules/no-whitespace-before-property.md [hunk 6]
```diff
@@ -10,7 +10,7 @@ foo. bar .baz . quz
 
 ## Rule Details
 
-This rule alerts for whitespace around the dot or before the opening bracket before properties of objects if they are on the same line. It does not alert for whitespace when the object and property are on separate lines, as it is common to add newlines to longer chains of properties:
+This rule disallows whitespace around the dot or before the opening bracket before properties of objects if they are on the same line. It does not alert for whitespace when the object and property are on separate lines, as it is common to add newlines to longer chains of properties:
```
Comment (pedrottimark) [inline docs/rules/no-whitespace-before-property.md]: s/It does not alert/It allows/

File: docs/rules/object-curly-newline.md [hunk 7]
```diff
@@ -7,34 +7,26 @@ A number of style guides require or disallow line breaks inside of object braces
 ## Rule Details
 
 This rule enforces consistent line breaks inside braces.
```
Comment (pedrottimark) [inline docs/rules/object-curly-newline.md]: inside braces of object literals or destructuring assignments.

File: docs/rules/object-curly-newline.md [hunk 8]
```diff
@@ -7,34 +7,26 @@ A number of style guides require or disallow line breaks inside of object braces
...
 ## Options
 
-```json
-{
-    "object-curly-newline": ["error", {"multiline": true}]
-}
-```
-
-This rule has options of 4 kinds:
+This rule has a string option:
```
Comment (pedrottimark) [inline docs/rules/object-curly-newline.md]: This rule has either a string option:

File: docs/rules/object-curly-newline.md [hunk 9]
```diff
@@ -7,34 +7,26 @@ A number of style guides require or disallow line breaks inside of object braces
...
 
-* `"always"` - requires line breaks always.
-* `"never"` - disallows line breaks.
-* `{multiline: true}` (default) - requires line breaks if there are line breaks inside properties or between properties. Otherwise, disallows line breaks.
-* `{minProperties: <integer>}` - requires line breaks if the number of properties is more than the given integer. Otherwise, disallows line breaks.
+* `"always"` - requires line breaks inside braces
+* `"never"` - disallows line breaks inside braces
 
-`multiline` and `minProperties` can be combined.
+This rule has an object option:
```
Comment (pedrottimark) [inline docs/rules/object-curly-newline.md]: Or an object option:

File: docs/rules/object-curly-newline.md [hunk 10]
```diff
@@ -7,34 +7,26 @@ A number of style guides require or disallow line breaks inside of object braces
...
 
-* `{multiline: true, minProperties: <integer>}` - requires line breaks if there are line breaks inside properties or between properties, or if the number of properties is more than the given integer. Otherwise, disallows line breaks.
+* `"multiline": true` (default) - requires line breaks if there are line breaks inside properties or between properties. Otherwise, disallows line breaks.
+* `"minProperties"` - requires line breaks if the number of properties is more than the given integer. Otherwise, disallows line breaks.
 
 Also, we can separate configuration for each object literal and destructuring assignment:
```
Comment (pedrottimark) [inline docs/rules/object-curly-newline.md]: You can specify different options for object literals and destructuring assignments:

**Comment by pedrottimark** [comment]
@scriptdaemon Thank you. Nice work. 5 rules was a good amount for now.

In `object-curly-newline`
* I suggested some sentences. According to my notes, you will see more of this in the next few batches: `one-var`, `padded-blocks`, `space-before-function-paren` so we can learn as we go.
* Can you delete the hyphens in the list items under Options so they read like simple sentences
* Just to confirm the reasoning (which makes sense to me) you deleted the **multiline and minProperties** section because `"multiline": true` is the default option, there that example code is redundant with (and almost but not quite identical to) the example code in the preceding **minProperties** section?

File: docs/rules/object-curly-newline.md [hunk 11]
```diff
@@ -322,82 +314,12 @@ let {k = function() {
...
-    j
-} = obj;
-let {
-    k = function() {
-        dosomething();
-    }
-} = obj;
-```
-
 ### separating configuration
```
Comment (pedrottimark) [inline docs/rules/object-curly-newline.md]: An alternative to consider for this heading is **ObjectExpression and ObjectPattern**

File: docs/rules/no-unneeded-ternary.md (hunk 4 above)
Comment (scriptdaemon) [inline docs/rules/no-unneeded-ternary.md]: Ninja vanish!

**Comment by scriptdaemon** [comment]
@pedrottimark Thank you. Yeah, I figured 5 was better than none.

The examples for combined multiline and minProperties did seem redundant to me, so that is why I removed them.

File: docs/rules/no-trailing-spaces.md (hunk 1 above)
Comment (pedrottimark) [inline docs/rules/no-trailing-spaces.md]: Oh, sorry for making the second suggestion in an ambiguous way. We ended up with two similar sentences in the paragraph.

**Comment by pedrottimark** [comment]
Only one more change in `no-trailing-spaces` where my line comment was ambiguous.

**Comment by scriptdaemon** [comment]
@pedrottimark Fixed. Probably should have paid more attention when making the latest updates. :smile:

**Comment by pedrottimark** [comment]
Looks good to me

**Pull request merged**